src/main/java/edu/kis/powp/jobs2d/events/MouseDrawListener.java

```diff
@@ -0,0 +1,40 @@
+package edu.kis.powp.jobs2d.events;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DriverFeature;
+
+import javax.swing.*;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+
+public class MouseDrawListener implements MouseListener {
+
+    private JPanel panel;
+
+    public MouseDrawListener(JPanel panel) {
+        this.panel = panel;
+    }
+
+    @Override
+    public void mouseClicked(MouseEvent e) {
+        Job2dDriver currentDriver = DriverFeature.getDriverManager().getCurrentDriver();
```

Comment:
Avoid a dependency to DriverFeature. You can just keep a reference to the manager.